oauthenticator/github.py

```diff
@@ -99,10 +106,14 @@ def authenticate(self, handler):
         resp = yield http_client.fetch(req)
         resp_json = json.loads(resp.body.decode('utf8', 'replace'))
         
-        username = resp_json["login"]
-        if self.whitelist and username not in self.whitelist:
-            username = None
-        return username
+        github_username = resp_json["login"]
+        #remap gihub username to system username
+        nix_username = self.username_map.get(github_username, github_username)
+
+        #check system username against whitelist
+        if self.whitelist and nix_username not in self.whitelist:
```

Comment:
There's ambiguity here. I would expect the whitelist to be applied *before* transforming the name to the system user.